### PR TITLE
Pass real MeterProvider to components

### LINKED ISSUE #6127: [jaeger-v2] Add Meter Provider To Telemetry Settings
### Requirement

We're currently using no-op meter providers in the code instead of propagating the provider from OTEL. To change this, we need to do the following: 
- [ ] Add OTEL's meter provider to https://github.com/jaegertracing/jaeger/blob/main/pkg/telemetery/settings.go#L16-L22 
- [ ] Propagate this setting to places where `noopmetric.NewMeterProvider()` is used

### PR DESCRIPTION
**Pull request by chahatsagarmain**
## Which problem is this PR solving?
- Fixes #6127 

## How was this change tested?
- make lint test

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by chahatsagarmain** [comment]
@mahadzaryab1 how will i propogate leveledmeterprovider where `telset ` is not part of function arguement ? Do i just use noop meterprovider there ?

**Comment by mahadzaryab1** [comment]
> @mahadzaryab1 how will i propogate leveledmeterprovider where `telset ` is not part of function arguement ? Do i just use noop meterprovider there ?

Yes, it will not be available in the context of v1 so a noop provider can be used there.

File: cmd/jaeger/internal/extension/jaegerquery/server.go
Comment (mahadzaryab1) [inline cmd/jaeger/internal/extension/jaegerquery/server.go:63]: i don't believe you need to do this - you should just be able to pass down the meter provider that you get from OTEL's  telemetry settings

File: pkg/telemetery/settings.go
Comment (mahadzaryab1) [inline pkg/telemetery/settings.go:24]:
```suggestion
	LeveledMeterProvider func(configtelemetry.Level) metric.MeterProvider
```

**Review by mahadzaryab1 — COMMENTED** [review]
nice work - just a couple of nits

File: cmd/jaeger/internal/extension/jaegerquery/server.go
Comment (chahatsagarmain) [reply cmd/jaeger/internal/extension/jaegerquery/server.go]: I dont think i need to do that here , All the test are working fine with just passing it down .

**Review by mahadzaryab1 — COMMENTED** [review]
lgtm but need to wait for @yurishkuro's approval

**Comment by yurishkuro** [comment]
I don't think this is addressing all places
```
cmd/query/app/server.go
22:	"go.opentelemetry.io/otel/metric/noop"

plugin/storage/grpc/factory.go
21:	noopmetric "go.opentelemetry.io/otel/metric/noop"

cmd/collector/app/handler/otlp_receiver.go
24:	noopmetric "go.opentelemetry.io/otel/metric/noop"

cmd/collector/app/handler/zipkin_receiver.go
16:	noopmetric "go.opentelemetry.io/otel/metric/noop"
```

**Comment by mahadzaryab1** [comment]
@yurishkuro (1) looks to be addressed in this PR. (3) and (4) are v1 components so we're passing in noop (https://github.com/jaegertracing/jaeger/blob/main/cmd/collector/app/handler/otlp_receiver.go#L71-L81). Is there anything else we can do for that? 

@chahatsagarmain for (3) We can pass the meter provider down from [this](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerstorage/extension.go#L129) constructor and into the grpc factory.

File: plugin/storage/grpc/factory.go
Comment (mahadzaryab1) [inline plugin/storage/grpc/factory.go:243]: you can remove the todo comment above this

**Review by mahadzaryab1 — COMMENTED** [review]
just one more comment

File: plugin/storage/grpc/factory.go
Comment (yurishkuro) [inline plugin/storage/grpc/factory.go:69]: Use the opportunity to simplify the signature:
```suggestion
	host component.Host,
	telset telemetry.Settings,
```

File: pkg/telemetery/settings.go
Comment (yurishkuro) [inline pkg/telemetery/settings.go:24]:
```suggestion
	Metrics              metrics.Factory
	LeveledMeterProvider func(configtelemetry.Level) metric.MeterProvider
	TracerProvider       trace.TracerProvider
	ReportStatus         func(*componentstatus.Event)
	Host                 component.Host
```

File: plugin/storage/grpc/factory.go
Comment (yurishkuro) [inline plugin/storage/grpc/factory.go:68]: I meant Jaeger's internal `telemetry.Settings`, which includes both host and metrics factory

Comment (chahatsagarmain) [reply plugin/storage/grpc/factory.go:68]: `telset := telemetery.Setting{
				Logger: s.telset.Logger,
				Host: host,
				Metrics: mf,
				LeveledMeterProvider: func(_ configtelemetry.Level)  metric.MeterProvider {
					return s.telset.MeterProvider
				 },
			}`

Comment (chahatsagarmain) [reply plugin/storage/grpc/factory.go:68]: can i pass meter provider like this  ? or should i have a field to pass meter provider in function signature ?

Comment (yurishkuro) [reply plugin/storage/grpc/factory.go:68]: yes that's fine (ie to ignore the level)

**Comment by chahatsagarmain** [comment]
@yurishkuro spm test is failing for recent prs

**Comment by mahadzaryab1** [comment]
> @yurishkuro spm test is failing for recent prs

Opened a PR for the fix: https://github.com/jaegertracing/jaeger/pull/6217

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks

**Pull request merged**
